### PR TITLE
Feature/issue 1 typescript UI rewrite

### LINKED ISSUE #1: TECHNICAL DEBT - Typescript UI Rewrite

### PR DESCRIPTION
**Pull request by mads5606**

### CODE REVIEW

File: interface/src/components/dialogs/newcontainer_dialog/NewContainerDialog.component.tsx [hunk 1]
```diff
@@ -90,14 +120,22 @@ function NewContainerDialog({ open, handleClose, dialogTitle, servers }: any) {
...
         <DialogContent>
           <form noValidate autoComplete="off">
             <Grid container spacing={2}>
-              <Grid item xs={12} style={{ padding: "2px", paddingLeft: "12px" }}>
-                <Typography variant="subtitle2">Server to Start Container On</Typography>
+              <Grid
+                item
+                xs={12}
+                style={{ padding: "2px", paddingLeft: "12px" }}
+              >
```
Comment (dstoft) [inline interface/src/components/dialogs/newcontainer_dialog/NewContainerDialog.component.tsx]: I'm not sure that I agree that this has to be exploded into more lines.

File: interface/src/components/dialogs/newcontainer_dialog/NewContainerDialog.component.tsx [hunk 2]
```diff
@@ -76,9 +104,11 @@ function NewContainerDialog({ open, handleClose, dialogTitle, servers }: any) {
         volumesFrom: [],
         networkMode: "",
       });
-      return
+      return;
     }
-    var name: any = Object.keys(templates).find((template) => template === templateName);
+    var name: any = Object.keys(templates).find(
+      (template) => template === templateName
+    );
```
Comment (dstoft) [inline interface/src/components/dialogs/newcontainer_dialog/NewContainerDialog.component.tsx]: Unnecessary?

File: interface/src/components/dialogs/newcontainer_dialog/NewContainerDialog.component.tsx [hunk 3]
```diff
@@ -52,14 +72,22 @@ function NewContainerDialog({ open, handleClose, dialogTitle, servers }: any) {
...
+      }
+    );
     console.log(JSON.stringify(valuesToSend));
     if (socketConnection !== undefined)
-      socketConnection.invoke(CREATE_NEW_CONTAINER_REQUEST, selectedServer, JSON.stringify(valuesToSend));
+      socketConnection.invoke(
+        CREATE_NEW_CONTAINER_REQUEST,
+        selectedServer,
+        JSON.stringify(valuesToSend)
+      );
```
Comment (dstoft) [inline interface/src/components/dialogs/newcontainer_dialog/NewContainerDialog.component.tsx]: Unnecessary?

File: interface/src/components/dialogs/newcontainer_dialog/NewContainerDialog.component.tsx [hunk 4]
```diff
@@ -33,9 +49,13 @@ function NewContainerDialog({ open, handleClose, dialogTitle, servers }: any) {
     volumesFrom: [],
     networkMode: "",
   });
-  const [selectedServer, setSelectedServer] = React.useState<string>(servers[0].url);
+  const [selectedServer, setSelectedServer] = React.useState<string>(
+    servers[0].url
+  );
```
Comment (dstoft) [inline interface/src/components/dialogs/newcontainer_dialog/NewContainerDialog.component.tsx]: Unnecessary?

File: interface/src/components/dialogs/newcontainer_dialog/NewContainerDialog.component.tsx [hunk 5]
```diff
@@ -33,9 +49,13 @@ function NewContainerDialog({ open, handleClose, dialogTitle, servers }: any) {
...
   const [selectedTemplate, setSelectedTemplate] = React.useState<string>("");
-  const socketConnection: HubConnection | undefined = useSelector((store: IRootState) => store.containerData.socketConnection);
+  const socketConnection: HubConnection | undefined = useSelector(
+    (store: IRootState) => store.containerData.socketConnection
+  );
```
Comment (dstoft) [inline interface/src/components/dialogs/newcontainer_dialog/NewContainerDialog.component.tsx]: Unnecessary?

File: interface/src/components/dialogs/newcontainer_dialog/NewContainerDialog.component.tsx [hunk 6]
```diff
@@ -120,8 +160,14 @@ function NewContainerDialog({ open, handleClose, dialogTitle, servers }: any) {
                   ))}
                 </TextField>
               </Grid>
-              <Grid item xs={12} style={{ padding: "2px", paddingLeft: "12px" }}>
-                <Typography variant="subtitle2">Choose a Template Here</Typography>
+              <Grid
+                item
+                xs={12}
+                style={{ padding: "2px", paddingLeft: "12px" }}
+              >
```
Comment (dstoft) [inline interface/src/components/dialogs/newcontainer_dialog/NewContainerDialog.component.tsx]: Unnecessary?

File: interface/src/components/dialogs/newcontainer_dialog/NewContainerDialog.component.tsx [hunk 7]
```diff
@@ -135,7 +181,9 @@ function NewContainerDialog({ open, handleClose, dialogTitle, servers }: any) {
                   type="text"
                   fullWidth
                   value={selectedTemplate}
-                  onChange={(event: any) => handleAutoFillTemplate(event.target.value)}
+                  onChange={(event: any) =>
+                    handleAutoFillTemplate(event.target.value)
+                  }
```
Comment (dstoft) [inline interface/src/components/dialogs/newcontainer_dialog/NewContainerDialog.component.tsx]: Unnecessary?

File: interface/src/components/dialogs/newcontainer_dialog/NewContainerDialog.component.tsx [hunk 8]
```diff
@@ -145,7 +193,11 @@ function NewContainerDialog({ open, handleClose, dialogTitle, servers }: any) {
                   ))}
                 </TextField>
               </Grid>
-              <Grid item xs={12} style={{ padding: "2px", paddingLeft: "12px" }}>
+              <Grid
+                item
+                xs={12}
+                style={{ padding: "2px", paddingLeft: "12px" }}
+              >
```
Comment (dstoft) [inline interface/src/components/dialogs/newcontainer_dialog/NewContainerDialog.component.tsx]: Unnecessary?

File: interface/src/components/dialogs/newcontainer_dialog/NewContainerDialog.component.tsx [hunk 9]
```diff
@@ -224,7 +280,11 @@ function NewContainerDialog({ open, handleClose, dialogTitle, servers }: any) {
               />
 
               <Grid container alignItems="center" justify="center">
-                <Grid item xs={10} style={{ padding: "2px", paddingLeft: "12px" }}>
+                <Grid
+                  item
+                  xs={10}
+                  style={{ padding: "2px", paddingLeft: "12px" }}
+                >
```
Comment (dstoft) [inline interface/src/components/dialogs/newcontainer_dialog/NewContainerDialog.component.tsx]: Unnecessary?

File: interface/src/components/dialogs/newcontainer_dialog/NewContainerDialog.component.tsx [hunk 10]
```diff
@@ -239,7 +299,10 @@ function NewContainerDialog({ open, handleClose, dialogTitle, servers }: any) {
                         });
                       }}
                     >
-                      <RemoveIcon color="primary" aria-label="delete port binding" />
+                      <RemoveIcon
+                        color="primary"
+                        aria-label="delete port binding"
+                      />
```
Comment (dstoft) [inline interface/src/components/dialogs/newcontainer_dialog/NewContainerDialog.component.tsx]: Unnecessary?

File: interface/src/components/dialogs/newcontainer_dialog/NewContainerDialog.component.tsx [hunk 11]
```diff
@@ -249,7 +312,10 @@ function NewContainerDialog({ open, handleClose, dialogTitle, servers }: any) {
                       onClick={() => {
                         setValues({
                           ...values,
-                          ports: [...values.ports, { containerPort: "", hostPort: "" }],
+                          ports: [
+                            ...values.ports,
+                            { containerPort: "", hostPort: "" },
+                          ],
```
Comment (dstoft) [inline interface/src/components/dialogs/newcontainer_dialog/NewContainerDialog.component.tsx]: Unnecessary?

File: interface/src/components/dialogs/newcontainer_dialog/NewContainerDialog.component.tsx [hunk 12]
```diff
@@ -307,8 +373,14 @@ function NewContainerDialog({ open, handleClose, dialogTitle, servers }: any) {
               ))}
 
               <Grid container alignItems="center" justify="center">
-                <Grid item xs={10} style={{ padding: "2px", paddingLeft: "12px" }}>
-                  <Typography variant="subtitle2">Environment Variables</Typography>
+                <Grid
+                  item
+                  xs={10}
+                  style={{ padding: "2px", paddingLeft: "12px" }}
+                >
```
Comment (dstoft) [inline interface/src/components/dialogs/newcontainer_dialog/NewContainerDialog.component.tsx]: Unnecessary?

File: interface/src/components/dialogs/newcontainer_dialog/NewContainerDialog.component.tsx [hunk 13]
```diff
@@ -322,7 +394,10 @@ function NewContainerDialog({ open, handleClose, dialogTitle, servers }: any) {
                         });
                       }}
                     >
-                      <RemoveIcon color="primary" aria-label="delete environment variable" />
+                      <RemoveIcon
+                        color="primary"
+                        aria-label="delete environment variable"
+                      />
```
Comment (dstoft) [inline interface/src/components/dialogs/newcontainer_dialog/NewContainerDialog.component.tsx]: Unnecessary?

File: interface/src/components/dialogs/newcontainer_dialog/NewContainerDialog.component.tsx [hunk 14]
```diff
@@ -332,11 +407,17 @@ function NewContainerDialog({ open, handleClose, dialogTitle, servers }: any) {
                       onClick={() => {
                         setValues({
                           ...values,
-                          environment: [...values.environment, { key: "", value: "" }],
+                          environment: [
+                            ...values.environment,
+                            { key: "", value: "" },
+                          ],
```
Comment (dstoft) [inline interface/src/components/dialogs/newcontainer_dialog/NewContainerDialog.component.tsx]: Unnecessary?

File: interface/src/components/dialogs/newcontainer_dialog/NewContainerDialog.component.tsx [hunk 15]
```diff
@@ -332,11 +407,17 @@ function NewContainerDialog({ open, handleClose, dialogTitle, servers }: any) {
...
                         });
                       }}
                     >
-                      <AddIcon color="primary" aria-label="add environment variable" />
+                      <AddIcon
+                        color="primary"
+                        aria-label="add environment variable"
+                      />
```
Comment (dstoft) [inline interface/src/components/dialogs/newcontainer_dialog/NewContainerDialog.component.tsx]: Unnecessary?

File: interface/src/components/dialogs/newcontainer_dialog/NewContainerDialog.component.tsx [hunk 16]
```diff
@@ -388,7 +469,11 @@ function NewContainerDialog({ open, handleClose, dialogTitle, servers }: any) {
               ))}
 
               <Grid container alignItems="center" justify="center">
-                <Grid item xs={10} style={{ padding: "2px", paddingLeft: "12px" }}>
+                <Grid
+                  item
+                  xs={10}
+                  style={{ padding: "2px", paddingLeft: "12px" }}
+                >
```
Comment (dstoft) [inline interface/src/components/dialogs/newcontainer_dialog/NewContainerDialog.component.tsx]: Unnecessary?

**Review by dstoft — CHANGES_REQUESTED** [review]
I really like the changes, it will make it way easier to maintain.
Except for a few lines in "NewContainerDialog.component.tsx", you have exploded a few lines into several lines, where I disagree on the readability improvement. I believe it may have been caused by your IDE attempting to enforce a maximum line length.
Normally I would not comment on this, but since this PR will probably decide how our code looks in the future, we should probably address it.
What is your opinion on this? 

Such as:
> (OLD) const [selectedServer, setSelectedServer] = React.useState<string>(servers[0].url);
> (NEW) const [selectedServer, setSelectedServer] = React.useState<string>(
> (NEW) servers[0].url
> (NEW) );

File: interface/src/components/dialogs/newcontainer_dialog/NewContainerDialog.component.tsx (hunk 1 above)
Comment (mads5606) [reply interface/src/components/dialogs/newcontainer_dialog/NewContainerDialog.component.tsx]: Agreed, no need to.

File: interface/src/components/dialogs/newcontainer_dialog/NewContainerDialog.component.tsx (hunk 2 above)
Comment (mads5606) [reply interface/src/components/dialogs/newcontainer_dialog/NewContainerDialog.component.tsx]: Agreed

File: interface/src/components/dialogs/newcontainer_dialog/NewContainerDialog.component.tsx (hunk 3 above)
Comment (mads5606) [reply interface/src/components/dialogs/newcontainer_dialog/NewContainerDialog.component.tsx]: Agreed

File: interface/src/components/dialogs/newcontainer_dialog/NewContainerDialog.component.tsx (hunk 4 above)
Comment (mads5606) [reply interface/src/components/dialogs/newcontainer_dialog/NewContainerDialog.component.tsx]: Agreed.

File: interface/src/components/dialogs/newcontainer_dialog/NewContainerDialog.component.tsx (hunk 5 above)
Comment (mads5606) [reply interface/src/components/dialogs/newcontainer_dialog/NewContainerDialog.component.tsx]: Agreed

File: interface/src/components/dialogs/newcontainer_dialog/NewContainerDialog.component.tsx (hunk 6 above)
Comment (mads5606) [reply interface/src/components/dialogs/newcontainer_dialog/NewContainerDialog.component.tsx]: Agreed

File: interface/src/components/dialogs/newcontainer_dialog/NewContainerDialog.component.tsx (hunk 7 above)
Comment (mads5606) [reply interface/src/components/dialogs/newcontainer_dialog/NewContainerDialog.component.tsx]: Agreed

File: interface/src/components/dialogs/newcontainer_dialog/NewContainerDialog.component.tsx (hunk 8 above)
Comment (mads5606) [reply interface/src/components/dialogs/newcontainer_dialog/NewContainerDialog.component.tsx]: Agreed

File: interface/src/components/dialogs/newcontainer_dialog/NewContainerDialog.component.tsx (hunk 9 above)
Comment (mads5606) [reply interface/src/components/dialogs/newcontainer_dialog/NewContainerDialog.component.tsx]: Agreed

File: interface/src/components/dialogs/newcontainer_dialog/NewContainerDialog.component.tsx (hunk 10 above)
Comment (mads5606) [reply interface/src/components/dialogs/newcontainer_dialog/NewContainerDialog.component.tsx]: Agreed

File: interface/src/components/dialogs/newcontainer_dialog/NewContainerDialog.component.tsx (hunk 11 above)
Comment (mads5606) [reply interface/src/components/dialogs/newcontainer_dialog/NewContainerDialog.component.tsx]: Agreed

File: interface/src/components/dialogs/newcontainer_dialog/NewContainerDialog.component.tsx (hunk 12 above)
Comment (mads5606) [reply interface/src/components/dialogs/newcontainer_dialog/NewContainerDialog.component.tsx]: Agreed

File: interface/src/components/dialogs/newcontainer_dialog/NewContainerDialog.component.tsx (hunk 13 above)
Comment (mads5606) [reply interface/src/components/dialogs/newcontainer_dialog/NewContainerDialog.component.tsx]: Agreed

File: interface/src/components/dialogs/newcontainer_dialog/NewContainerDialog.component.tsx (hunk 14 above)
Comment (mads5606) [reply interface/src/components/dialogs/newcontainer_dialog/NewContainerDialog.component.tsx]: Agreed

File: interface/src/components/dialogs/newcontainer_dialog/NewContainerDialog.component.tsx (hunk 15 above)
Comment (mads5606) [reply interface/src/components/dialogs/newcontainer_dialog/NewContainerDialog.component.tsx]: Agreed

File: interface/src/components/dialogs/newcontainer_dialog/NewContainerDialog.component.tsx (hunk 16 above)
Comment (mads5606) [reply interface/src/components/dialogs/newcontainer_dialog/NewContainerDialog.component.tsx]: Agreed

**Comment by mads5606** [comment]
> I really like the changes, it will make it way easier to maintain.
> Except for a few lines in "NewContainerDialog.component.tsx", you have exploded a few lines into several lines, where I disagree on the readability improvement. I believe it may have been caused by your IDE attempting to enforce a maximum line length.
> Normally I would not comment on this, but since this PR will probably decide how our code looks in the future, we should probably address it.
> What is your opinion on this?
> 
> Such as:
> 
> > (OLD) const [selectedServer, setSelectedServer] = React.useState(servers[0].url);
> > (NEW) const [selectedServer, setSelectedServer] = React.useState(
> > (NEW) servers[0].url
> > (NEW) );

I completely agree, the IDE is a bit over-zealous. Much prefer keeping it single-line when possible, especially when its a single expression, and not an encapsulation of several statements. I have changed the formatting to reflect it.

**Review by dstoft — APPROVED** [review]

**Pull request merged**